### PR TITLE
pydantic 2.8.2 (new formula)

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

package inclusions, https://repology.org/project/pydantic/versions
would help remove rust deps for lots of python formulae

### CODE REVIEW

File: Formula/p/pydantic.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  def python3
+    which("python3.12")
+  end
+
+  def install
+    venv = virtualenv_create(libexec, python3)
+    venv.pip_install resources
+    venv.pip_install_and_link buildpath
+    site_packages = Language::Python.site_packages(python3)
+    (prefix/site_packages/"homebrew-pydantic.pth").write venv.site_packages
```
Comment (branchvincent) [inline Formula/p/pydantic.rb:37]: This pollutes the global site packages with more than just `pydantic`, which we really shouldn't continue doing

Alternatively, (and in fact, I think we should be doing this across the board) I suggest making this `keg_only`. Our own formulae will still be able to find it due to `brew` automatically creating [`.pth` files for dependents](https://github.com/Homebrew/brew/blob/99e2d93ea5c6e9f8543e962d43949e7b8b06b772/Library/Homebrew/language/python.rb#L194)

Also, before we do this I think we need to finally add a `pip check` somewhere in `brew` to catch incompatible requirements. There's [several packages ](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-core%20%22pydantic-1%22&type=code) that are pinned to `pydantic<2` and we need to ensure we don't break them

But I'm interested in other's thoughts, cc @Homebrew/core

Comment (MikeMcQuaid) [reply Formula/p/pydantic.rb:37]: > This pollutes the global site packages with more than just `pydantic`, which we really shouldn't continue doing

Agree both in this case and that this should be something we audit for.

**Comment by daeho-ro** [comment]
Is it possible to upgrade `rust` from `1.79.0_1` to `1.80.1` on MacOS 12 (x86_64)?

I tried to reproduce the `maturin` build error on AWS with intel based MacOS 12.7 and got error per 2-3 attempts.
```
cargo(12799,0x10eea0600) malloc: *** error for object 0x600003663e20: pointer being freed was not allocated
cargo(12799,0x10eea0600) malloc: *** set a breakpoint in malloc_error_break to debug
zsh: abort      cargo metadata --manifest-path Cargo.toml --format-version 1 --locked
```
But this is not the same error on the CI process where it shows the error code -6.

Anyway, maybe this can be gone with the most recent version of `rust`, so I am just curious about are there any way to upgrade the brew package versions on the CI process.

**Comment by p-linnane** [comment]
@daeho-ro We just need to rebase to have CI pick up the latest Rust update.

**Comment by daeho-ro** [comment]
Nice, the error is gone!

**Review by p-linnane — DISMISSED** [review]

**Comment by branchvincent** [comment]
I think we need https://github.com/Homebrew/homebrew-core/pull/178715/files#r1694038551 to be addressed

**Review by MikeMcQuaid — COMMENTED** [review]
Agreed on needing comment addressed

**Pull request closed without merging**